docs/NEW_LANGUAGE_FEATURE_INTEGRATION_PROCESS.md

```diff
@@ -0,0 +1,153 @@
+# New Language Feature Check Integration Process [WIP]
+
+The ability to parse new language features must be implemented and
+merged before following this document. After that, we need to update
+existing checks and create new checks to support the new language features.
+This document outlines the procedures for integrating new language features into Checkstyle.
+It includes guidelines for how to discover new checks and
+existing checks that need updates to ensure that Checkstyle remains
+aligned with language enhancements.
+
+## Check Update Procedure
+
+Updating check modules to support new language features aligns Checkstyle
+with current best practices in the Java community
+and contributes to the project's ongoing evolution.
+
+### Consider Similar and Related Tokens
+
+Identify existing tokens that are similar to the new language feature tokens.
+This comparison helps to determine which existing checks may be relevant
+and require updates to support the new token. Additionally,
+we should consider the impact that a new child token may have
+on existing checks.
+
+**Examples**:
+
+- When the `RECORD_DEF` token was introduced, it was reasonable to look at all checks
+  that had `CLASS_DEF` in their acceptable tokens.
+
+- When the `TEXT_BLOCK_CONTENT` token was introduced, it was reasonable to look at all checks
+  that had `STRING_LITERAL` in their acceptable tokens.
+
+- When switch expressions were introduced, it was reasonable to look at all checks
+  that deals with expressions because `LITERAL_SWITCH` may now fall under `EXPR` token.
+  This requires analyzing the impact of this new child token on existing checks,
+  such as `InnerAssignmentCheck`
+
+### Java Enhancement Proposals (JEPS)
```

Comment:
Let's make this one first, as @rnveach already alluded to, we should be reading JEPs before anything else.

Reply:
@nrmancuso Are you saying we don't need the issue anymore? https://github.com/checkstyle/checkstyle/issues/14923

Reply:
Let’s keep the issue and revisit it later in the summer.